File: ruby/nth-prime/nth_prime.rb
module Prime
  class << self
    def nth(number)
      raise ArgumentError, "Argument should be greater than 0" if number < 1

      primes = [2, 3]
      loop do
        primes << find_next_prime(primes) 
        return primes[number - 1] if primes.size >= number
      end
    end

    private
    
    def find_next_prime(primes) 
      next_number = primes.last
      loop do
        next_number += 2
        prime_check = !primes.any? do |prime|
          next_number % prime == 0
        end

        return next_number if prime_check
      end
    end
  end
end

File: python/nth-prime/nth_prime.py
def _is_prime(primes, n):
    for x in primes:
        if n % x == 0:
            return False

    return True

def nth_prime(number):
    primes = [2]
    
    n = 3
    while len(primes) != number:
        if _is_prime(primes, n):
            primes.append(n)

        n += 2

    return primes[-1]
